### PR TITLE
Improve batch docs

### PR DESCRIPTION
**Pull request by LuKks**
Thanks @HDegroote the initial explanation: https://github.com/holepunchto/hyperbee/pull/90/commits/08a3fcac5422ea32f6149d3983479e8579f5439b
I didn't even know that we had a lock method lol

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -168,6 +168,14 @@ Make a new atomic batch that is either fully processed or not processed at all.
 
 If you have several inserts and deletions then a batch can be much faster.
 
+Batch is snapshotted at creation. Outside writes are not taken into account when reading.
```
Comment (LuKks) [inline README.md:171]: Not highly true:
```js
const batch = await db.batch()
t.is(batch.version, 1)
await db.put('/a')
t.is(batch.version, 2)
console.log(await batch.get('/a')) // => { seq, key, value }
```

So state is not snapshotted?

Comment (mafintosh) [reply README.md:171]: A put (obvs) updates the snap.

File: README.md [hunk 2]
```diff
@@ -168,6 +168,14 @@ Make a new atomic batch that is either fully processed or not processed at all.
...
+
+#### `await batch.lock()`
+
+Force acquire the single write lock of the Hyperbee.
+
+Otherwise, it's auto-acquired when you do the first `put` or `del` in a batch.
```
Comment (mafintosh) [inline README.md:177]: Dont document this, no semver guaranteees on this

File: README.md (hunk 1 above)
Comment (LuKks) [reply README.md:171]: It's a `db.put` external write outside of batch, is that intended to update the snap of the batch?

Comment (mafintosh) [reply README.md:171]: Ah db my bad. I'd prefer to leave this be for now. Its due to update being called. This needs some (breaking) refactor to remove all update calls

**Pull request merged**